src/Application/Commands/DeleteShoppingList/DeleteShoppingListCommand.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Lindyhopchris\ShoppingList\Application\Commands\DeleteShoppingList;
+
+use Lindyhopchris\ShoppingList\Persistance\Json\JsonFileHandler;
+
+class DeleteShoppingListCommand implements DeleteShoppingListCommandInterface
+{
+    /**
+     * @var JsonFileHandler
+     */
+    private JsonFileHandler $fileHandler;
+
+    /**
+     * @param JsonFileHandler $fileHandler
+     */
+    public function __construct(JsonFileHandler $fileHandler)
```

Comment:
Required: this change to the constructor dependency isn't right I'm afraid. The purpose of the repository interface is that it decouples the commands from a _specific_ storage implementation. By injecting the `JsonFileHandler` here, your command will only ever work with the JSON storage. If we switched to using a database storage, then this command wouldn't work.

This morning we talked about implementing a `delete` method on the repository interface. You need to stick to that approach. Internally, the JSON storage (that implements the interface) would then delete the file via the file handler.